liger_iris_pipeline/pipeline/imager_stage2.py

```diff
@@ -42,57 +42,22 @@ class ImagerStage2Pipeline(LigerIRISPipeline):
 
     def process(self, input):
 
-        # Load the association
-        self.asn = self.input_to_asn(input)
-
-        self.log.info(f"Starting ImagerStage2Pipeline ...")
-
         # Each exposure is a product in the association.
         # Process each exposure.
-        results = []
-        for product in self.asn["products"]:
-            result = self.process_exposure_product(product)
-            results.append(result)
-
-        self.log.info("ImagerStage2Pipeline completed")
-
-        return results
-    
-    # Process each exposure
-    def process_exposure_product(self, exp_product : dict):
-        """Process an exposure product.
-
-        Parameters:
-            exp_product (dict): The exposure product.
-        """
-
-        # Members by type
-        members_by_type = self.asn_product_by_types(exp_product)
-
-        # Get the science member. Assumes only one
-        # TODO: Enforce one science exposure per product in Association class
-        science = members_by_type["sci"][0]
-
+        science = input["SCI"]
         with self.open_model(science) as input_model:
-            
-            self.log.info(f"Processing {science}")
-
-            # Run remaining steps
             input_model = self.parse_subarray_map.run(input_model)
             input_model = self.dark_sub.run(input_model)
             input_model = self.flat_field.run(input_model)
-            if len(members_by_type["sky"]) > 0:
-                sky_filename = members_by_type["sky"][0]
-                input_model = self.sky_sub.run(input_model, sky=sky_filename)
+            if "SKY" in input:
+                input_model = self.sky_sub.run(input_model, sky=input['SKY'])
```

Comment:
Do we always want to execute this code? It you so to provider a way to disable this?